flow/record/adapter/gcs.py

```diff
@@ -0,0 +1,88 @@
+import logging
+import re
+from fnmatch import fnmatch
+from typing import Iterator, Union
+
+from google.cloud.storage.client import Client
+from google.cloud.storage.fileio import BlobReader, BlobWriter
+
+from flow.record.adapter import AbstractReader, AbstractWriter
+from flow.record.base import Record, RecordAdapter
+from flow.record.selector import CompiledSelector, Selector
+
+__usage__ = """
+Google Cloud Storage adapter
+---
+Read usage: rdump gcs://[PROJECT-ID]:[BUCKET-ID]?path=[PATH]
```

Comment:
Could you change `BUCKET-ID` to `BUCKET_ID` and same for `PROJECT_ID`?

Also, what are your thoughts of making the `gcs://` uri more `gsutil` like compatible?

For example. getting rid of the `path` parameter and access it like:

`gcs://bucket-id/path/`

And an example with specifying the `project-id`:

`gcs://my-fancy-bucket/path/to/records/?project_id=my-project-12345`

Reply:
This also has the added benefit that you could just access it like this:

```python
with RecordReader("gcs://my-fancy-bucket/records/*.avro", project_id="my-project-12345") as reader:
    for record in reader:
        print(record)